server/public/model/metrics.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package model
+
+import (
+	"fmt"
+	"time"
+)
+
+type MetricType string
+
+const (
+	ClientMetricChannelVisited MetricType = "channel_visited"
+	ClientMetricChannelLoad    MetricType = "channel_load"
+
+	performanceReportTTLMilliseconds = 300 * 1000 // 300 seconds/5 minutes
+)
+
+type MetricSample struct {
+	Metric    MetricType        `json:"metric"`
+	Value     int64             `json:"value"`
+	Timestamp int64             `json:"timestamp,omitempty"`
+	Labels    map[string]string `json:"labels,omitempty"`
+}
+
+// PerformanceReport is a set of samples collected from a client
+type PerformanceReport struct {
+	Version    string            `json:"version"`
```

Comment:
Are we going to validate this for now or is any value for it fine?

Reply:
Yeah we are going to validate it, I'll add a validation code for the `PerformanceReport`.